UI/obs-app.cpp

```diff
@@ -1510,6 +1510,34 @@ static uint64_t convert_log_name(bool has_prefix, const char *name)
 	return std::stoull(timestring.str());
 }
 
+/* If upgrading from an older (non-XDG) build of OBS, move config files to XDG directory. */
+#if defined(__FreeBSD__)
```

Comment:
I really dislike adding boilerplate code for a single version upgrade that - as experience has taught me - will then linger on as dead code for years, especially as it perpetuates the patchwork of our function bodies (there really should be a single "InitPlatform()` method on `OBSApp` that uses a platform-independent API and uses platform-dependent source files instead of littering our functions with preprocessor conditions all over the place, it's just messy all around).

That said I'm fine with this as a quick fix if a TODO is removed with a version target so that this will be removed after a majority of users has probably been automatically migrated and a commitment to remove it then.

Reply:
You're correct, this code was originally introduced in 2015 in ba02e065fe91 and persisted until 2023 (removed in c0532168243b78a7f880b3ef0c1b20cfe9be51ca).

How about keeping it for two major versions? I'll add `/* TODO: Remove after version 32.0. */`

Reply:
Yeah I kinda thought about keeping it for a full major version cycle as well.